src/nccl_ofi_rdma.c

```diff
@@ -98,7 +98,6 @@ static size_t eager_max_size = 0;
 /* List of comms undergoing deferred cleanup */
 static nccl_ofi_deque_t *s_comm_cleanup_list = NULL;
 static nccl_ofi_deque_t *r_comm_cleanup_list = NULL;
-static pthread_mutex_t comm_cleanup_list_lock = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
This is a bit confusing but I think you'll still need this one? The cleanup list is at process level, not a domain level.

Two threads could simultaneously close two different communicators, and they'll both try to add their communicators to these cleanup lists.